adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,525 @@
+/**
+ * Licensed to Big Data Genomics (BDG) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The BDG licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.adam.hbase
+
+import java.util
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable
+import org.apache.hadoop.hbase.io.encoding.DataBlockEncoding
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.hadoop.hbase.client._
+import org.apache.hadoop.hbase._
+import org.apache.hadoop.hbase.spark._
+import org.bdgenomics.adam.rdd.variation.{ GenotypeRDD, VariantContextRDD }
+import org.bdgenomics.adam.rich.RichVariant
+import org.bdgenomics.formats.avro._
+import org.apache.hadoop.hbase.spark.HBaseRDDFunctions._
+import org.apache.hadoop.hbase.util.Bytes
+import org.apache.hadoop.hbase.{ HBaseConfiguration, TableName }
+import org.apache.avro.specific.SpecificDatumWriter
+import org.apache.avro.io._
+import java.io.ByteArrayOutputStream
+import org.apache.hadoop.hbase.io.compress.Compression.Algorithm
+import org.apache.avro.specific.SpecificDatumReader
+import org.apache.hadoop.hbase.mapreduce.LoadIncrementalHFiles
+import org.apache.hadoop.fs.{ FSDataInputStream, FileSystem, Path }
+import org.bdgenomics.adam.models.{ ReferenceRegion, ReferencePosition, SequenceDictionary, VariantContext }
+import scala.io.Source
+
+import scala.collection.mutable.ListBuffer
+import sys.process._
+
+import scala.collection.JavaConverters._
+
+/**
+ * Created by jp on 7/23/16.
+ */
+
+object HBaseFunctions {
+
+  def saveSequenceDictionaryToHBase(hbaseTableName: String,
+                                    sequences: SequenceDictionary,
+                                    sequenceDictionaryId: String): Unit = {
+
+    val conf = HBaseConfiguration.create()
+    val connection = ConnectionFactory.createConnection(conf)
+    val table: Table = connection.getTable(TableName.valueOf(hbaseTableName))
+    val contigs = sequences.toAvro
+
+    val baos: java.io.ByteArrayOutputStream = new ByteArrayOutputStream()
```

Comment:
Can you doc this section of the code? If I read this right, you are making an avro encoder that writes bytes into an array, so that you can then write Avro records into this array, which you then copy to HBase?